src/components/application_manager/rpc_plugins/sdl_rpc_plugin/test/commands/hmi/subscribe_button_response_test.cc

```diff
@@ -0,0 +1,112 @@
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include <string>
+
+#include "gtest/gtest.h"
+#include "hmi/subscribe_button_response.h"
+
+namespace test {
+namespace components {
+namespace commands_test {
+namespace hmi_commands_test {
+namespace subscribe_button_response {
+
+namespace am = ::application_manager;
+using application_manager::ExpiredButtonRequestData;
+using application_manager::commands::ResponseFromHMI;
+using sdl_rpc_plugin::commands::hmi::SubscribeButtonResponse;
+using ::test::components::event_engine_test::MockEventDispatcher;
+typedef std::shared_ptr<ResponseFromHMI> ResponseFromHMIPtr;
+
+namespace {
+const uint32_t kCorrelationId = 2u;
+const uint32_t kAppId = 1u;
+const hmi_apis::Common_ButtonName::eType kButtonName =
+    hmi_apis::Common_ButtonName::SEEKLEFT;
+}  // namespace
+
+class HMISubscribeButtonResponseTest
+    : public CommandsTest<CommandsTestMocks::kIsNice> {
+ public:
+  MockEventDispatcher event_dispatcher_;
+
+  HMISubscribeButtonResponseTest() {
+    ON_CALL(app_mngr_, event_dispatcher())
+        .WillByDefault(ReturnRef(event_dispatcher_));
+  }
+};
+
+TEST_F(HMISubscribeButtonResponseTest, RUN_NotEmptyExpiredData) {
+  MessageSharedPtr command_msg(CreateMessage(smart_objects::SmartType_Map));
+  (*command_msg)[strings::params][strings::correlation_id] = kCorrelationId;
+  (*command_msg)[am::strings::params][am::hmi_response::code] =
+      hmi_apis::Common_Result::SUCCESS;
+
+  ResponseFromHMIPtr command(
+      CreateCommand<SubscribeButtonResponse>(command_msg));
+
+  ExpiredButtonRequestData expired_data;
+  expired_data.app_id_ = kAppId;
+  expired_data.button_name_ = kButtonName;
+  utils::Optional<ExpiredButtonRequestData> expired_data_opt = expired_data;
+
+  ON_CALL(app_mngr_, GetExpiredButtonRequestData(kCorrelationId))
+      .WillByDefault(Return(expired_data_opt));
+  ON_CALL(mock_message_helper_,
+          CreateButtonSubscriptionHandlingRequestToHmi(_, _, _, _))
+      .WillByDefault(Return(command_msg));
+  EXPECT_CALL(mock_rpc_service_, SendMessageToHMI(command_msg));
+
+  command->Run();
+}
+
+TEST_F(HMISubscribeButtonResponseTest, RUN_EmptyExpiredData) {
+  MessageSharedPtr command_msg(CreateMessage(smart_objects::SmartType_Map));
+  (*command_msg)[strings::params][strings::correlation_id] = kCorrelationId;
+  (*command_msg)[am::strings::params][am::hmi_response::code] =
+      hmi_apis::Common_Result::SUCCESS;
+  ResponseFromHMIPtr command(
+      CreateCommand<SubscribeButtonResponse>(command_msg));
+
+  ON_CALL(app_mngr_, GetExpiredButtonRequestData(kCorrelationId))
+      .WillByDefault(Return(utils::Optional<ExpiredButtonRequestData>::EMPTY));
+
+  EXPECT_CALL(event_dispatcher_, raise_event(_));
+  command->Run();
+}
+
```

Comment:
Also suggest to add one more important negative case: if we have expired request, but result of its execution is not successful. In this case we don't need to send hmi request and don't need to process event